cmake/modules/platform/os/zos.cmake

```diff
@@ -21,7 +21,7 @@
 
 list(APPEND OMR_PLATFORM_DEFINITIONS
 	-D_ALL_SOURCE
-	-D_OPEN_THREADS=2
+	-D_OPEN_THREADS=3
```

Comment:
why the move from 2 to 3 is needed? can we document the justification and effects of non-trivial changes in the commit message?

Reply:
Sure, I will add the details to the commit description as well.
After having encountered some compilation issues related to `pthread` methods, the team working on Open XL pointed me towards https://www.ibm.com/docs/en/zos/2.5.0?topic=files-feature-test-macros, and heavily suggested to change to `OPEN_THREADS=3` in order to enable/expose more pthread functions used.